### PR TITLE
string.getObject to get array element by numeric index

### PR DESCRIPTION
**Pull request by ilyavf**
To work with arrays or `DefineList`s:

```
var list = new DefineList([1,2,3])
string.getObject(0, list);  // >>> 1
```

### CODE REVIEW

**Comment by ilyavf** [comment]
instead of 

```
var parts = "" + name ? ("" + name).split('.') : [],
```

it could be smth like:

```
var parts = typeof name === 'number' ? [name] : (name ? name.split('.') : []),
```

I think the current variant is simpler, and both currently work, but I am not sure if we have to handle name as undefined or null.

File: js/string/string.js [hunk 1]
```diff
@@ -88,7 +88,7 @@ var string = {
 	getObject: function (name, roots, add) {
 		// The parts of the name we are looking up
 		// `['App','Models','Recipe']`
-		var parts = name ? name.split('.') : [],
+		var parts = "" + name ? ("" + name).split('.') : [],
```
Comment (justinbmeyer) [inline js/string/string.js]: Looks good.  Thinking about this, it's possible, but unlikely, that someone could try to get the `""` property.

```
var obj = {"": "empty"};
getObject("",[obj]);
```

I think we should account for this.

**Comment by justinbmeyer** [comment]
What about

``` js
var parts = name != null ? ("" + name).split('.') : [],
```

**Comment by ilyavf** [comment]
@justinbmeyer Good! I like it. I made the change and rebased to the latest master. Tests are passing.

**Pull request merged**